src/Microsoft.Health.Dicom.Core/Features/CustomTag/ICustomTagStore.cs

```diff
@@ -21,6 +22,21 @@ public interface ICustomTagStore
         /// <returns>The tag key.</returns>
         Task<long> AddCustomTagAsync(string path, string vr, CustomTagLevel level, CustomTagStatus status, CancellationToken cancellationToken = default);
 
+        /// <summary>
+        /// Get a requested custom tag from CustomTagStore by tagPath.
+        /// </summary>
+        /// <param name="path">Path associated with requested custom tag formatted as it is stored internally.</param>
+        /// <param name="cancellationToken">The cancellation token.</param>
+        /// <returns>Custom tag entry with path, VR, level and status.</returns>
+        Task<CustomTagEntry> GetCustomTagAsync(string path, CancellationToken cancellationToken = default);
```

Comment:
I'm thinking on ICustomTagStore level, could we merge these 2 functions together?
We can do something like https://github.com/microsoft/dicom-server/blob/368a21c761c83a14288bec80dc2146ccd7c2f539/src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Migrations/1.sql#L629  to handle Null path

Reply:
I want to keep them separate till sql so that the return values are distinct!

---
In reply to: [558596602](https://github.com/microsoft/dicom-server/pull/501#discussion_r558596602) [](ancestors = 558596602)